src/pylhe/__init__.py

```diff
@@ -190,11 +190,13 @@ def read_lhe_init(filepath):
                 for child in element:
                     # Find all weightgroups
                     if child.tag == "weightgroup" and child.attrib != {}:
-                        try:
+                        if "type" in child.attrib:
                             wg_type = child.attrib["type"]
-                        except KeyError:
+                        elif "name" in child.attrib:
+                            wg_type = child.attrib["name"]
```

Comment:
This is a PR that should have gotten addressed months ago (sorry) so let me see if I can write some tests for this this week to cover this. If not, we'll merge this in and then have to write tests in the future.